frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -82,6 +83,7 @@ public void modelChanged(Model model) throws WorkerInitializationException {
     private void addThreads(List<WorkerThread> threads, Model model, int count)
             throws WorkerInitializationException {
         int maxGpu = configManager.getNumberOfGpu();
+        UUID threadUuid = UUID.randomUUID();
```

Comment:
This should be in for loop, otherwise all the worker has the same name,
Actually, each worker thread already has a unique name, you can just call thread.getName() to retrieve it.
And I believe your old way using threadId should work. You was using a wrong thread Id in addJob() call.

Reply:
Replaced this with getName().